core/conn/jdbcT4/src/main/java/org/trafodion/jdbc/t4/T4DatabaseMetaData.java

```diff
@@ -3092,27 +3090,40 @@ public boolean dataDefinitionIgnoredInTransactions() throws SQLException {
 			// dropped from the search."
 			procedureNamePattern = "%";
 
+		}else{
+			procedureNamePattern = sqlValueFormat(procedureNamePattern);
+		}
+		MetadataHolder mdHolder = this.connection_.mdHolder_;
+		int retryCnt = 0;
+		while(retryCnt < 3 ) {
+			try {
+				PreparedStatement ps = mdHolder.getPStatement(SQL_API_SQLPROCEDURES, this.connection_);
+				ps.setObject(1, schemaPattern);
+				ps.setObject(2, procedureNamePattern);
+				return ps.executeQuery();
+			} catch (SQLException e) {
+				mdHolder.removePStatement(SQL_API_SQLPROCEDURES);
+				++retryCnt;
+				if( retryCnt >= 3 ) {
+					throw e;
+				}
+			}
 		}
-		getSQLCatalogsInfo(connection_.getServerHandle(), // Server Handle
-				SQL_API_SQLPROCEDURES, // catalogAPI
-				catalogNm, // catalog
-				schemaNm, // schema
-				procedureNamePattern, // table name
-				"", // tableTypeList
-				"", // cloumn name
-				(int) 0, // cloumnType
-				(int) 0, // rowIdScope
-				(long) 0, // nullable
-				(int) 0, // uniqueness
-				(int) 0, // accuracy
-				(short) 0, // sqlType
-				(int) 0, // metadataId
-				"", // fcatalog
-				"", // fschema
-				"" // ftable
-		);
+		return null;
+	}
```

Comment:
getSQLCatalogsInfo is used for all catalog APIs. It looks like the change is to do it differently for getProcedures and getProcedureColumns. Can you please provide some context about the need for this change.